src/RA_Dlg_Memory.cpp

```diff
@@ -1040,9 +1048,19 @@ INT_PTR Dlg_Memory::MemoryProc(HWND hDlg, UINT nMsg, WPARAM wParam, LPARAM lPara
                     EnableWindow(GetDlgItem(hDlg, IDC_RA_RESULTS_BACK), TRUE);
                     EnableWindow(GetDlgItem(hDlg, IDC_RA_RESULTS_FORWARD), FALSE);
 
-                    unsigned int nValueQuery = 0;
+                    SearchResult& srPrevious = *(m_SearchResults.end() - 2);
+                    SearchResult& sr = m_SearchResults.back();
+                    sr.m_nCompareType = nCmpType;
 
+                    if (IsDlgButtonChecked(hDlg, IDC_RA_CBO_GIVENVAL) == BST_UNCHECKED)
                     {
+                        sr.m_results.Initialize(srPrevious.m_results, nCmpType);
```

Comment:
pass in the previous result set for a list of addresses to check, and their previous values